### PR TITLE
Pt sim election

### PR DESCRIPTION
**Pull request by guitarvydas**
It appears that faked elections work.  I believe that there are enough comments in src/election.lisp and node-sim.lisp.

### CODE REVIEW

File: src/Cosi-BLS/cosi-handlers.lisp [hunk 1]
```diff
@@ -1316,10 +1316,12 @@ bother factoring it with NODE-COSI-SIGNING."
 
 ;; -------------------------------------------------------------
 
+#|
```
Comment (easye) [inline src/Cosi-BLS/cosi-handlers.lisp:1319]: @guitarvydas Just go ahead and remove such references to a specific network configuration.

File: src/sim-election.lisp [hunk 2]
```diff
@@ -0,0 +1,138 @@
...
+(defvar *beacon-interval*  7)
+(defvar *self-destruct*    nil)
+
+(defun kill-beacon ()
+  (setf *self-destruct* t))
+
+;; mp is the multi-processing library for LispWorks
+(defun make-trial-election-beacon (handler-actor)
+  (unless *beacon-timer*
+    (setf *beacon-timer* (mp:make-timer (lambda ()
```
Comment (easye) [inline src/sim-election.lisp:43]: @guitarvydas We have to make the code work under `ccl` (and hopefully soon `sbcl`).  I think you should use the `mp-compat` system here to make a timer, or go find a suitable system in Quicklisp

**Review by easye — CHANGES_REQUESTED** [review]

File: src/node-sim.lisp [hunk 3]
```diff
@@ -31,9 +31,21 @@ witnesses."
               (not (and (probe-file cosi-simgen::*default-data-file*)
                         (probe-file cosi-simgen::*default-key-file*))))
     (cosi-simgen::generate-tree :nodes nodes))
-  (setf cosi-simgen::*cosi-prepare-timeout* cosi-prepare-timeout)
```
Comment (easye) [inline src/node-sim.lisp:34]: @guitarvydas You've deleted the configuration options to set the Cosi timeouts.  Please reinstate, or explain why you deleted this option.

**Review by easye — CHANGES_REQUESTED** [review]
1.  Elections are happening too fast for anything meaningful.  Please allow the configuration of the time between elections.

2.  Holding an election without sealing a block doesn't make sense.  When the election timer fires, the block should be sealed, then the election held.  As such, you should probably deprecate the `force 

3.  Please output the identity of the new leader after the election has occurred.

**Comment by dbmcclain** [comment]
Elections should be run by all witness nodes. There is no need to broadcast the outcome of the leader election. If, perchance, two different nodes think they both won the election, only one - at most - can achieve BFT.

**Pull request closed without merging**